tests/build/test_launch_script.py

```diff
@@ -137,6 +138,31 @@ def test_successful_lora():
         assert os.path.exists(tempdir + "/adapter_config.json") is True
 
 
+def test_successful_lora_target_modules_defaults():
+    """Check if target modules automatically set to their default values"""
```

Comment:
Please be more specific about what this means. Something like: "Check that if target_modules is not set, or set to None, the default value by model type will be using in training". And explain the test:
"We use `TRAIN_KWARGS` to run training with, which has no `target_modules` set. During HF training process, the correct default target modules will be used for model type Llama and "q_proj", "v_proj" will then exist in the resulting in adapter_config.json."

And link to https://github.com/huggingface/peft/blob/7b1c08d2b5e13d3c99b7d6ee83eab90e1216d4ba/src/peft/tuners/lora/model.py#L432 for how it is handled in HF

Reply:
It seems like the [lora test](https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/tests/test_sft_trainer.py#L352) in test_sft_trainer.py are already testing default target_modules being None, with additional testing for specific configs. Should I still add the above test?

Reply:
I have a feeling that this test [linked](https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/tests/test_sft_trainer.py#L381) is bypassing the argument parsing from CLI or from JSON and just calling the `train` function directly with the None argument for target_modules. 

We want to test that even when user passes None or not passing anything from CLI or from JSON, it still works. Like the test you  wrote for this PR. 
(The problem with the test you wrote for this PR is it tests the accelerate launch script only, so someone who calls `tuning.sft_trainer` in cli will bypass your test as they're not calling the accelerate launch script. Hence I requested for the test to be moved to sft_trainer). 

I see that there's no testing of `main` in `test_sft_trainer`, this looks like it would be first!